pages/pil/exemptions/views/modules.jsx

```diff
@@ -1,60 +1,138 @@
-import React from 'react';
+import React, { Component, Fragment } from 'react';
 import mapKeys from 'lodash/mapKeys';
 import { connect } from 'react-redux';
+import { Button } from '@ukhomeoffice/react-components';
 import {
   Snippet,
   FormLayout,
   Inset,
   Fieldset,
-  Header,
-  AddAnother
+  Header
 } from '@asl/components';
-import { Select } from '@ukhomeoffice/react-components';
-import { species } from '@asl/constants';
 
-import { normalise } from '../../../../lib/utils';
+import { species } from '@asl/constants';
 
-const SPECIES_REVEAL_TOTAL_COUNT = 10;
-const SPECIES_REVEAL_VISIBLE_COUNT = 1;
+const content = require('../content/modules');
 
 const connectComponent = key => {
-  const mapStateToProps = ({ model, static: { schema, errors } }) => {
+  const mapStateToProps = ({ model, static: { schema, errors, modulesThatRequireSpecies } }) => {
     schema = schema.modules.options.find(m => m.value === key).reveal;
-    return {
+    const props = {
       model,
       errors,
-      schema: mapKeys(schema, (v, k) => `module-${key}-${k}`)
+      option: key,
+      schema: mapKeys(schema, (v, k) => `module-${key}-${k}`),
+      modulesThatRequireSpecies
     };
+
+    return props;
   };
 
-  return connect(mapStateToProps)(Fieldset);
+  return connect(mapStateToProps)(RepeatedFieldset);
+};
+
+const speciesOptions = {
+  inputType: 'select',
+  options: species,
+  label: content.fields.species.label
 };
 
+class RepeatedFieldset extends Component {
+
+  constructor(options) {
+    super(options);
+
+    this.state = {
+      items: this.props.model[`module-${this.props.type}-species`]
```

Comment:
this looks like it could be quite fragile - could we use `get`:
```js
this.state = {
  items: get(this.props, `model[module-${this.props.type}-species]`, [])
}
```

Reply:
done